### ISSUE TITLE
Blade directives are missed

### ISSUE THREAD

**Issue by ihoru** [xinax/laravel-gettext#34]
I have a custom directive installed in Blade in my project:
```php
$gettext_callback = function ($expression) {
    return '<?php echo e(_'.$expression.'); ?>';
};
\Blade::directive('_', $gettext_callback);
```

I would like to create a short alias in templates in order to write this:
```
@_('translate me')
```
instead of this:
```
{{ _('translate me') }}
```

But in compiled files (storage/framework/messages/{md5}.php) my directive does not work. It happens because new $compile object is created for each source path.

If class BladeCompiler would have public method getCustomDirectives, I could solve this problem and send you pull request, but it does not.

Can you suggest me something? I can also try to send pull request with getCustomDirectives added to BladeCompiler sources...

**Comment by ihoru** [comment]
I've created pull-request, waiting for response:
https://github.com/laravel/framework/pull/9800

**Comment by ihoru** [comment]
> https://github.com/laravel/framework/pull/9800
merged to 5.1

**Comment by shaggyz** [comment]
@ihoru Great work. Thank you.
